tf_files/aws/modules/eks-nodepool/cloud.tf

```diff
@@ -0,0 +1,342 @@
+  associate_public_ip_address = false
+  iam_instance_profile        = "${aws_iam_instance_profile.eks_node_instance_profile.name}"
+  image_id                    = "${data.aws_ami.eks_worker.id}"
+  instance_type               = "${var.instance_type}"
+  name_prefix                 = "eks-${var.vpc_name}-nodepool-${var.nodepool}"
+  security_groups             = ["${aws_security_group.eks_nodes_sg.id}", "${aws_security_group.ssh.id}"]
+  user_data_base64            = "${base64encode(data.template_file.bootstrap.rendered)}"
+  key_name                    = "${var.ec2_keyname}"
+
+  root_block_device {
+    volume_size = 30
+  }
+
+
+  lifecycle {
+    create_before_destroy = true
+    #ignore_changes  = ["user_data_base64"]
+  }
+}
+
+
+resource "aws_autoscaling_group" "eks_autoscaling_group" {
+  desired_capacity     = 2 
+  launch_configuration = "${aws_launch_configuration.eks_launch_configuration.id}"
+  max_size             = 10
+  min_size             = 2 
+  name                 = "eks-${var.nodepool}worker-node-${var.vpc_name}"
+  #vpc_zone_identifier  = ["${data.aws_subnet.eks_private.*.id}"]
+  #vpc_zone_identifier  = ["${data.aws_subnet_ids.private.ids}"]
+  vpc_zone_identifier  = ["${var.eks_private_subnets}"]
+
+  tag {
+    key                 = "Environment"
+    value               = "${var.vpc_name}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Name"
+    value               = "eks-${var.vpc_name}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "kubernetes.io/cluster/${var.vpc_name}"
+    value               = "owned"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "k8s.io/cluster-autoscaler/enabled"
+    value               = ""
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "k8s.io/cluster-type/eks"
+    value               = ""
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "k8s.io/nodepool/${var.nodepool}"
+    value               = ""
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "k8s.io/cluster-autoscaler/node-template/label/role"
+    value               = "${var.nodepool}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "k8s.io/cluster-autoscaler/node-template/taint/role"
+    value               = "${var.nodepool}:NoSchedule"
+    propagate_at_launch = true
+  }
+
+# Avoid unnecessary changes for existing commons running on EKS 
+  lifecycle {
+    ignore_changes = ["desired_capacity","max_size","min_size"]
+  }
+}
+
+
+# Let's allow ssh just in case
+resource "aws_security_group" "ssh" {
+  name        = "ssh_eks_${var.vpc_name}-nodepool-${var.nodepool}"
+  description = "security group that only enables ssh"
+  vpc_id      = "${data.aws_vpc.the_vpc.id}"
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "TCP"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
```

Comment:
If I'm running a jupyter notebood in a pod on the cluster, can I ssh from there into a node?  No, because I still need to have a valid ssh key ...

Reply:
Workers do not store any private key and if you can upload one into the notebook, you still need the public one in the node you are trying to ssh in.